Sources/SwifterSwift/UIKit/UIImageExtensions.swift

```diff
@@ -32,6 +32,37 @@ public extension UIImage {
         return withRenderingMode(.alwaysTemplate)
     }
 
+    /// SwifterSwift: Average color for this image
+    #if canImport(CoreImage)
+    func averageColor() -> UIColor? {
+        // https://stackoverflow.com/questions/26330924
+        guard let ciImage = CIImage(image: self) else {
+            return nil
+        }
+
+        // CIAreaAverage returns a single-pixel image that contains the average color for a given region of an image.
+        let parameters = [kCIInputImageKey: ciImage, kCIInputExtentKey: CIVector(cgRect: ciImage.extent)]
+        guard let outputImage = CIFilter(name: "CIAreaAverage", parameters: parameters)?.outputImage else {
+            return nil
+        }
+
+        // After getting the single-pixel image from the filter extract pixel's RGBA8 data
+        var bitmap = [UInt8](repeating: 0, count: 4)
+        let context = CIContext(options: [.workingColorSpace: NSNull()])
```

Comment:
Why do you set this color space? The documentation says
> To request that Core Image perform no color management, specify the NSNull object as the value for this key. Use this option for images that don’t contain color data (such as elevation maps, normal vector maps, and sampled function tables).

but this image may have a color space, no? I.e. `cgImage.colorSpace`. So maybe here you could use
`[.workingColorSpace: cgImage.colorSpace ?? NSNull()]`

Reply:
good idea